packages/react-sdk/src/components/BackgroundFilters/BackgroundFilters.tsx

```diff
@@ -0,0 +1,263 @@
+  BackgroundConfig,
+  createRenderer,
+  loadTFLite,
+  TFLite,
+} from '@stream-io/video-filters';
+
+export type BackgroundFiltersProps = {
+  isBlurringEnabled?: boolean;
+  backgroundImages?: string[];
+  backgroundFilter?: BackgroundConfig;
+  backgroundImage?: string;
+  backgroundBlurLevel?: BackgroundBlurLevel;
+  tfFilePath?: string;
+  modelFilePath?: string;
+  basePath?: string;
+};
+
+export type BackgroundFiltersAPI = {
+  disableBackgroundFilter: () => void;
+  applyBackgroundBlurFilter: (blurLevel: BackgroundBlurLevel) => void;
+  applyBackgroundImageFilter: (imageUrl: string) => void;
+};
+
+export type BackgroundFiltersContextValue = BackgroundFiltersProps &
+  BackgroundFiltersAPI;
+
+const BackgroundFiltersContext = createContext<
+  BackgroundFiltersContextValue | undefined
+>(undefined);
+
+export const useBackgroundFilters = () => {
+  const context = useContext(BackgroundFiltersContext);
+  if (!context) {
+    throw new Error(
+      'useBackgroundFilters must be used within a BackgroundFiltersProvider',
+    );
+  }
+  return context;
+};
+
+export const BackgroundFiltersProvider = (
+  props: PropsWithChildren<BackgroundFiltersProps>,
+) => {
+  const {
+    children,
+    isBlurringEnabled = true,
+    backgroundImages = [],
+    backgroundFilter: bgFilterFromProps = 'none',
+    backgroundImage: bgImageFromProps = undefined,
+    backgroundBlurLevel: bgBlurLevelFromProps = 'high',
+    tfFilePath,
+    modelFilePath,
+    basePath,
+  } = props;
+
+  const [backgroundFilter, setBackgroundFilter] = useState(bgFilterFromProps);
+  const [backgroundImage, setBackgroundImage] = useState(bgImageFromProps);
+  const [backgroundBlurLevel, setBackgroundBlurLevel] =
+    useState(bgBlurLevelFromProps);
+
+  const applyBackgroundImageFilter = useCallback((imageUrl: string) => {
+    setBackgroundFilter('image');
+    setBackgroundImage(imageUrl);
+  }, []);
+
+  const applyBackgroundBlurFilter = useCallback(
+    (blurLevel: BackgroundBlurLevel = 'high') => {
+      setBackgroundFilter('blur');
+      setBackgroundBlurLevel(blurLevel);
+    },
+    [],
+  );
+
+  const disableBackgroundFilter = useCallback(() => {
+    setBackgroundFilter('none');
+    setBackgroundImage(undefined);
+    setBackgroundBlurLevel('high');
+  }, []);
+
+  return (
+    <BackgroundFiltersContext.Provider
+      value={{
+        backgroundImage,
+        backgroundBlurLevel,
+        backgroundFilter,
+        disableBackgroundFilter,
+        applyBackgroundBlurFilter,
+        applyBackgroundImageFilter,
+        backgroundImages,
+        isBlurringEnabled,
+        tfFilePath,
+        modelFilePath,
+        basePath,
+      }}
+    >
+      {children}
+      <BackgroundFilters />
+    </BackgroundFiltersContext.Provider>
+  );
+};
+
+const BackgroundFilters = () => {
+  const call = useCall();
+  const { backgroundImage, backgroundFilter } = useBackgroundFilters() || {};
+  const [videoRef, setVideoRef] = useState<HTMLVideoElement>();
```

Comment:
```suggestion
  const [videoRef, setVideoRef] = useState<ElementRef<'video'> | null>(null);
```
I think all of these should be initiated with `null` as that's what's being set when the element de-renders. That way you can get rid of all the @ts-expect-error directives.